src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1533,29 +1556,23 @@ bool WindowImplX11::processEvent(XEvent& windowEvent)
     // - Discard both duplicated KeyPress and KeyRelease events when KeyRepeatEnabled is false
 
     // Detect repeated key events
-    // (code shamelessly taken from SDL)
     if (windowEvent.type == KeyRelease)
     {
-        // Check if there's a matching KeyPress event in the queue
-        XEvent nextEvent;
-        if (XPending(m_display))
+        // Find the next KeyPress event with matching keycode and time
+        std::deque<XEvent>::iterator iter = std::find_if(
```

Comment:
Is there a reason to call `find_if` if `m_keyRepeat` is false?

Reply:
I do think so, as it has to fulfill the two tasks mentioned at the function top. 

* Discard duplicated KeyRelease events when KeyRepeatEnabled is true
* Discard both duplicated KeyPress and KeyRelease events when KeyRepeatEnabled is false

When repeat is true it will now ignore the repeated KeyRelease event that X generates.

When repeat is false it will erase the repeated KeyPress event from the queue and ignore the repeated KeyRelease event.

Reply:
Sorry, I didn't read the code carefully 😸